client.go

```diff
@@ -517,22 +517,66 @@ func (c *Client) Whisper(username, text string) {
 	c.send(fmt.Sprintf("PRIVMSG #%s :/w %s %s", c.ircUser, username, text))
 }
 
-// Join enter a twitch channel to read more messages
-func (c *Client) Join(channel string) {
-	channel = strings.ToLower(channel)
-
-	// If we don't have the channel in our map AND we have an
-	// active connection, explicitly join before we add it to our map
+// Join enter a twitch channel to read more messages.
+// Any channels which could not be joined due to the maximum
+// message length being reached are returned in a slice.
```

Comment:
Our client should handle that for the user. The user should not care what the message limits of IRC are or TMI are, we abstract that. 

So no return value please.